slightly improve rich embed behavior

0x70b1a5 opened the pull request:
- [x] Fix embeds not showing up in Urbit chats (now e.g. [automated YAGPDB posts from other places](https://yagpdb.xyz/) will show up in Urbit as intended)
- [x] ~~Break GIF-only posts~~ 

~~I'm willing to fix GIFs in a future commit.~~ EDIT: GIFs have been fixed.

midsum-salrux commented on setup.py:
```diff
@@ -13,7 +13,7 @@
     url="https://github.com/midsum-salrux/faux",
     packages=setuptools.find_packages(),
-    install_requires=['quinnat', 'discord.py-self'],
+    install_requires=['quinnat', 'discord.py'],
```
@ryjm this reverts the `discord.py` to `discord.py-self` change. I think I missed this change in the original PR review. What was the intention here?

0x70b1a5 replied:
I missed that this had been reverted, sorry. I was unable to install discord.py-self at first, but that might be because only discord.py is available to pip3.5, and I'm still not sure what discord.py-self is.

0x70b1a5 replied:
After installing discord.py, the bot functioned as expected, anyway.

ryjm replied:
sorry about this - i used this so i could talk as myself (plz don't tell anyone) and it requires using your own discord token vs the bot token.

0x70b1a5 commented:
Sorry about the extra PR there. 

I changed the logic a bit and moved the KeyError handling to its own function. Since we seem to not know what kind of embed a message has when we get it, this was the regrettable solution. 

The `message_type` is also a bit of a magic string, but it's a lot easier to use it than trying to create a system of Universal Post Formatting - for now, anyway.

ryjm commented:
also thank god you've fixed so much of my spaghetti code

0x70b1a5 commented:
It seems in all this that I fixed GIFs as well. Huzzah!(?)

midsum-salrux commented on faux/faux.py:
```diff
@@ -51,71 +59,112 @@ async def on_message(self, message):
+        )
+        if len(matching_channels) == 0:
+            return
+        message_type = ''
```
https://docs.python.org/3/library/enum.html#functional-api

I think Enums would be a better way to handle this. Something like:

```
from enum import Enum
MessageType = Enum("MessageType", "PLAIN GIF TWITTER REDDIT DISCORD")
message_type = MessageType.PLAIN
```

0x70b1a5 replied:
sounds good; can you tell I don't use Python? :)

ryjm replied:
nice, another non python user contributing. the things we do for urbit...

midsum-salrux reviewed (approved):
LGTM
